extension/onboarding/onboardingController.jsx

```diff
@@ -0,0 +1,69 @@
+/* globals React, ReactDOM, settings */
+
+this.onboardingController = (function() {
+  const exports = {};
+  const { useState, useEffect } = React;
+  const onboardingContainer = document.getElementById("onboarding-container");
+  let isInitialized = false;
+  let userSettings;
+
+  exports.OnboardingController = function() {
+    const [optinViewAlreadyShown, setOptinViewShown] = useState(false);
+    const [permissionError, setPermissionError] = useState(null);
+
+    useEffect(() => {
+      if (!isInitialized) {
+        isInitialized = true;
+        init();
+      }
+    });
+
+    const init = async () => {
+      if (location.pathname.endsWith("onboard.html")) {
+        launchPermission();
+      }
+      const result = await settings.getSettingsAndOptions();
+      userSettings = result.settings;
+    };
+
+    const setOptinValue = async value => {
+      // TODO: Is this the right setting to update?
+      userSettings.collectAudio = value;
+      await settings.saveSettings(userSettings);
+    };
+
+    const launchPermission = async () => {
```

Comment:
This should only be called if they've already opted in or out (so we aren't giving them the permission dialog at the same time as the telemetry dialog)

So something like:

```
if (optinViewAlreadyShown) {
  launchPermission();
}
```